hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/BaseTokenHandler.java

```diff
@@ -279,8 +279,9 @@ private List<TokenRelation> createTokenRelsToAccount(
             }
 
             // Create the new token relation
-            final var isFrozen = token.hasFreezeKey() && token.accountsFrozenByDefault();
-            final var kycGranted = !token.hasKycKey();
+            boolean isTreasuryAccount = token.treasuryAccountId().equals(account.accountId());
```

Comment:
Nit: I prefer `Objects.equals()` because it allows either parameter to be `null`.

Reply:
Changed it to use `Objects.equals()`